felix/daemon/bootstrap_linux.go

```diff
@@ -0,0 +1,17 @@
+package daemon
+
+import (
+	log "github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/calico/felix/wireguard"
+	"github.com/projectcalico/calico/libcalico-go/lib/clientv3"
+)
+
+func bootstrapWireguard(ifaceName string, felixHostName string, v3Client clientv3.Interface) {
+	err := wireguard.BootstrapHostConnectivity(
+		ifaceName, felixHostName, v3Client)
+
+	if err != nil {
+		log.WithError(err).Info("couldn't bootstrap wireguard host connectivity")
```

Comment:
Not sure why you're handling this here instead of bubbling it up?  The main daemon code should decide whether to retry/ignore/restart.

What should it actually do depends on what happens if clearing the key fails.  Does the failure to connect to typha already cause felix to restart?

Reply:
The failed connect to typha is currently a Fatal error, so I think the whole thing restarts, yes